cpp/ql/src/semmle/code/cpp/rangeanalysis/RangeUtils.qll

```diff
@@ -81,6 +81,5 @@ predicate isReducibleCFG(Function f) {
 
 predicate backEdge(PhiInstruction phi, PhiOperand op) {
   phi.getAnOperand() = op and
-  phi.getBlock().dominates(op.getPredecessorBlock())
-  // TODO: identify backedges during IR construction
-}
\ No newline at end of file
+  phi.getBlock() = op.getPredecessorBlock().getBackEdgeSuccessor(_)
```

Comment:
I believe you also need to delete the `isReducibleCFG(i.getFunction()` conjunct in `RangeAnalysis.qll` line 563 in order for this to be fully applied.

Reply:
See https://github.com/Semmle/ql/pull/633#discussion_r242101980

Reply:
Done. That made the test produce a new (good) result, so it looks like it worked.